src/main/java/at/hannibal2/skyhanni/features/misc/FrogMaskDisplay.kt

```diff
@@ -0,0 +1,86 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getItemStack
+import at.hannibal2.skyhanni.utils.RegexUtils.matchAll
+import at.hannibal2.skyhanni.utils.RenderUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object FrogMaskDisplay {
+
+    private val config get() = SkyHanniMod.feature.misc
+
+    private var display = Renderable.verticalContainer(listOf())
+
+    private var region: Pair<String, SimpleTimeMark> = "" to SimpleTimeMark.farPast()
+
+    private val patternGroup = RepoPattern.group("misc.frogmask")
+    private val activeRegionPattern by patternGroup.pattern(
+        "description.active",
+        "Today's region: (?<region>.+)",
+    )
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+
+        config.frogMaskDisplayPos.renderRenderable(display, posLabel = "Frog Mask Display")
+    }
+
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        if (!isEnabled()) return
+
+        display = if (region.first.isNotEmpty()) updateDisplay() else Renderable.verticalContainer(listOf())
+
+        if (region.second.isInFuture()) return
+
+        val helmet = InventoryUtils.getHelmet() ?: return
+        if (helmet.getInternalName() != "FROG_MASK".asInternalName()) return
+
+        activeRegionPattern.matchAll(helmet.getLore().map { it.removeColor() }) {
```

Comment:
instead of removing the colour it is probably better to keep the colour and filter the colour codes out in the pattern, or you could use the colour that hypixel gives the region with in the display